### DISCUSSION TITLE
labgrid connection lost after idle a few time

### DISCUSSION THREAD

**Discussion by atline** [labgrid-project/labgrid#1378] [Q&A]
Lock the board and use the console to open the serial, sometimes if we leave the terminal there for some times, it will reported next with a low probability, what could be the possible reason? Thanks.

```
switch to partitions #0, OK
mmc1 is current device
UID: 0xbe231828 0xa44fcbac 0x314d2d82 0xa98b9c5a
flash target is MMC:1
Net:   eth0: ethernet@42890000, eth1: ethernet@428a0000 [PRIME]
Fastboot: Normal
Normal Boot
Hit any key to stop autoboot:  0 
u-boot=> 
u-boot=> Invalid argument
connection lost
``` 

A new message just now:

```
In:    serial
Out:   serial
Err:   serial

BuildInfo:
  - ELE firmware version 1.1.0-dd32b403

switch to partitions #0, OK
mmc1 is current device
UID: 0x9702ad0 0x894521c4 0x86e8599b 0x689daf62
flash target is MMC:1
Net:   eth0: ethernet@42890000, eth1: ethernet@428a0000 [PRIME]
Fastboot: Normal
Normal Boot
Hit any key to stop autoboot:  0
u-boot=>
u-boot=>
u-boot=> Got EOF from port
connection lost
```

**Comment by Bastian-Krause** [comment]
What serial resource are you using?

> Invalid argument
> connection lost

There are multiple paths in https://github.com/pengutronix/microcom/blob/15130f84c70a0be1566f59f8a4e2c9fac6354a59/mux.c#L120 that could lead to this condition. Have a look there.

> Got EOF from port
> connection lost

That happens when the serial device disappears and/or if ser2net on the exporter stops or if you lose network connectivity between microcom and ser2net. Have a look in the journal of the machine with the physical serial device connected to it. See also https://github.com/pengutronix/microcom/blob/main/mux.c#L146
